### PR TITLE
Stack Edit and Delete tests

### PR DESCRIPTION
**Pull request by sshveta**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit ba8ddf656dd9ac4dea4e5ce36e4369d8b87ffcb6:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c64a26de2e78b0973d86479d7348e294016f5a7f:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/tests/cloud/test_stack.py [hunk 1]
```diff
@@ -61,6 +62,10 @@ def test_resources_link(stack):
 
 def test_edit_tags(stack):
     stack.edit_tags("Cost Center *", "Cost Center 001")
+    row = sel.elements("//*[(self::th or self::td) and normalize-space(.)={}]/../.."
```
Comment (psav) [inline cfme/tests/cloud/test_stack.py]: I think that perhaps this functionality should move back into the previous file, but maybe under a method named, ```get_tags```. We should be avoiding having ```sel``` calls in tests, unless we can't really avoid it. But in my opinion, getting the tags is a legitimate method for a Stack.

File: cfme/cloud/stack.py [hunk 2]
```diff
@@ -49,11 +48,14 @@ def edit_tags(self, tag, value):
...
+        self.get_tags(tag, value)
+
+    def get_tags(self, tag, value):
         sel.force_navigate('clouds_stack', context={'stack': self})
         row = sel.elements("//*[(self::th or self::td) and normalize-space(.)={}]/../.."
                      "//td[img[contains(@src, 'smarttag')]]".format(quoteattr("My Company Tags")))
-        tag = sel.text(row).strip()
-        assert tag == "Cost Center: Cost Center 001"
+        company_tag = sel.text(row).strip()
+        assert company_tag == tag.replace(" *", "") + ": " + value
```
Comment (psav) [inline cfme/cloud/stack.py]: I think the assertion of the tags should happen in edit

**Comment by dajoRH** [comment]
Lint report for commit 4da989ea70268ae414c9f3d6a9c1e37fa35d7da2:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/cloud/stack.py (hunk 2 above)
Comment (psav) [inline cfme/cloud/stack.py]: So to clarify....... ```get_tags``` should only be "getting the tag value/s"
An assertion to ensure the tag was changed to the right value, needs to be in the edit method.

Comment (seandst) [inline cfme/cloud/stack.py]: Agreed.

Also, when doing the check, it should be a proper exception, assertions are ignored under some conditions. Also, this is a good candidate for string interpolation with `.format`:

```python
if company_tag != "{}: {}".format(tag.replace(" *", ""), value):
    raise something other than AssertionError here
```

**Comment by dajohnso** [comment]
@sshveta I also think we should be asserting that the tags display in the summary screen of the object as well in addition to making sure that they successully add on the tags screen

**Comment by dajoRH** [comment]
Lint report for commit 413e52d927189bc44a6513d511adffeeaba989ac:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/cloud/stack.py [hunk 3]
```diff
@@ -49,11 +49,16 @@ def edit_tags(self, tag, value):
                               'select_value': value},
              action=form_buttons.save)
         flash.assert_success_message('Tag edits were successfully saved')
+        company_tag = self.get_tags(tag, value)
+        if company_tag != "{}: {}".format(tag.replace(" *", ""), value):
+            raise CFMEException("{} ({}) tag is not assigned!".format(tag.replace(" *", ""), value))
+
+    def get_tags(self, tag, value):
```
Comment (psav) [inline cfme/cloud/stack.py]: This shouldn't need to take ```tag``` and ```value``` now.

**Comment by dajoRH** [comment]
Lint report for commit 82323acd1c7c23a48579594af0d8b4536975eea5:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 615c373d8b23bac80e0093a0fa09f843f43d597f:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**